venom/src/main/java/com/github/venom/service/VenomService.kt

```diff
@@ -49,11 +50,17 @@ internal class VenomService : Service() {
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
         when (intent?.action) {
             ACTION_CANCEL -> cancelAndTerminate()
-            ACTION_KILL -> launchDeath()
+            ACTION_KILL -> kill()
+            ACTION_RESTART -> launchDeath()
         }
         return START_NOT_STICKY
     }
 
+    private fun kill() {
+        stopSelf()
+        Process.killProcess(Process.myPid())
```

Comment:
Hey Simon! Thanks for the contribution

According to my testing, the new button still restarts the app. Perhaps, I'm missing something - let's try to figure it out.

Have you run the sample app to test this solution?
It looks like you are using the same function to kill the app as we do for the kill-restart scenario. Any difference?

Reply:
My bad, I tested it once and assumed it worked. Unfortunately the OS tries to be "smart" and restart the process if it detects an Activity was not correctly stopped.